gulpfile.js

```diff
@@ -37,137 +33,216 @@ var regex = {
 };
 
 function getPackage() {
-    delete require.cache[require.resolve('./package.json')];
-    return require('./package.json');
+    delete require.cache[require.resolve("./package.json")];
+    return require("./package.json");
 }
 
 function getBundleConfig() {
-    delete require.cache[require.resolve('./bundleconfig.json')];
-    return require('./bundleconfig.json');
+    delete require.cache[require.resolve("./bundleconfig.json")];
+    return require("./bundleconfig.json");
 }
 
 function merge(streams) {
     return streams.length ? mergestream(streams) : mergestream().end();
 }
 
-gulp.task('min', ['min:js', 'min:css', 'min:html']);
+gulp.task("min", ["min:js", "min:css", "min:html"]);
 
 function mapSources() {
-    return sourcemaps.mapSources(function (sourcePath, file) {
-        var sourceRootPathEndIndex = sourcePath.indexOf('assets');
+    return sourcemaps.mapSources(function(sourcePath, file) {
+        var sourceRootPathEndIndex = sourcePath.indexOf("assets");
         var sourceRootPath = sourcePath.substring(0, sourceRootPathEndIndex);
         // ../../../ for assets/static/bundle + ../ count of parent folders in real path
-        var relativeRootPath = '../'.repeat((sourceRootPath.match(new RegExp('/', "g")) || []).length + 3) + sourcePath.substring(sourceRootPathEndIndex);
+        var relativeRootPath =
+            "../".repeat(
+                (sourceRootPath.match(new RegExp("/", "g")) || []).length + 3
+            ) + sourcePath.substring(sourceRootPathEndIndex);
         return relativeRootPath;
     });
 }
 
-gulp.task('min:js', function () {
-    var tasks = getBundles(regex.js).map(function (bundle) {
-        return gulp.src(bundle.inputFiles, { base: '.' })
+gulp.task("min:js", function() {
+    var tasks = getBundles(regex.js).map(function(bundle) {
+        return gulp
+            .src(bundle.inputFiles, { base: "." })
             .pipe(sourcemaps.init())
             .pipe(mapSources())
             .pipe(concat(bundle.outputFileName))
             .pipe(uglify({ mangle: false }))
             .pipe(sourcemaps.write("."))
-            .pipe(gulp.dest('.'));
+            .pipe(gulp.dest("."));
     });
     return merge(tasks);
 });
 
-gulp.task('min:css', function () {
-    var tasks = getBundles(regex.css).map(function (bundle) {
-        return gulp.src(bundle.inputFiles, { base: '.' })
+// move all js files from bower packages to single file
+gulp.task("packJavaScript", function() {
+    return gulp
+        .src(
+            mainBowerFiles({
```

Comment:
I think we can use this npm https://www.npmjs.com/package/bower-main for task of distinguish between minified and normal  files